lib/stream_data.ex

```diff
@@ -136,6 +157,11 @@ defmodule StreamData do
   end
 
   # TODO: should this be exposed?
+  # QUESTION: In Elixir, we would call this function filter_map,
+  # and have it return {:cont, value} | :skip or {:ok, value} | :error.
+  # Maybe we should standardize this return type and make this public?
+  # I would also likely call the underlying rose tree operation filter_map.
```

Comment:
I don't think `filter_map` is a good name here though, because we're not mapping, we're *binding*. Those are different concepts in StreamData and mapping is only a particular case of binding. But the return value, for sure I think we should standardize with elixir. Preferences?

About making it public, we can but maybe we can wait, I don't know. Preferences?